### PR TITLE
Remove cache directory when install runs out of disk

### PR DESCRIPTION
**Pull request by ltilve**
When any of the pull or deploy processes during a flatpak install
hits a ENOSPC situation, the created temporary files at
~/.local/share/flatpak/system-cache were not deleted leaving
the disk full.

https://phabricator.endlessm.com/T15391

### CODE REVIEW

File: common/flatpak-dir.c [hunk 1]
```diff
@@ -2730,9 +2739,16 @@ flatpak_dir_pull_untrusted_local (FlatpakDir          *self,
   if (!repo_pull_one_untrusted (self->repo, remote_name, url,
                                 subdirs_arg ? (const char **)subdirs_arg->pdata : NULL,
                                 ref, checksum, progress,
-                                cancellable, error))
+                                cancellable, &my_error))
     {
+
+      /* Ensure that cache directory gets cleaned on OOD */
+      if (g_error_matches (my_error, G_IO_ERROR, G_FILE_ERROR_NOSPC)) {
```
Comment (cosimoc) [inline common/flatpak-dir.c]: Not sure this opening curly brace should be there; where does it get closed?

File: common/flatpak-dir.c [hunk 2]
```diff
@@ -2730,9 +2739,16 @@ flatpak_dir_pull_untrusted_local (FlatpakDir          *self,
...
+        flatpak_rm_rf (ostree_repo_get_path(self->repo), cancellable, NULL);
```
Comment (cosimoc) [inline common/flatpak-dir.c]: Missing space before paren

File: common/flatpak-dir.c [hunk 3]
```diff
@@ -5064,8 +5080,17 @@ flatpak_dir_install (FlatpakDir          *self,
                                                    (const char * const *) subpaths,
                                                    installation ? installation : "",
                                                    cancellable,
-                                                   error))
+                                                   &my_error)) {
+
+        /* Ensure that we clean cache directory on OOD */
+        if (g_error_matches(my_error, G_IO_ERROR, G_FILE_ERROR_NOSPC)) {
```
Comment (cosimoc) [inline common/flatpak-dir.c]: Same comment above about the opening curly

**Comment by ltilve** [comment]
Thanks for the review. I have corrected the accidental curly brace leftovers and the missing space issue.

File: common/flatpak-dir.c [hunk 4]
```diff
@@ -5483,6 +5508,7 @@ flatpak_dir_update (FlatpakDir          *self,
       g_autofree char *url = NULL;
       gboolean gpg_verify_summary;
       gboolean gpg_verify;
+      g_autoptr(GError) my_error = NULL;
```
Comment (cosimoc) [inline common/flatpak-dir.c]: Don't think this is needed

**Comment by cosimoc** [comment]
One minor comment, looks good otherwise.

**Pull request merged**